drivers/riello_usb.c

```diff
@@ -143,7 +143,7 @@ static int Send_USB_Packet(uint8_t *send_str, uint16_t numbytes)
 		err = usb_bulk_write(udev, 0x2, (usb_ctrl_charbuf) USB_buff_pom, 8, 1000);
 
 		if (err < 0) {
-			upsdebugx(3, "USB: Send_USB_Packet: send_usb_packet, err = %d %s ", err, strerror(errno));
```

Comment:
I think the 3 hits in this file are not right: `err` is the return from `usb_bulk_write()` and probably returns the library's codes. While OS-based `errno` (hence `upsdebug_with_errno()` may add interesting context to troubleshooting, the acute problem would be diagnosed by `nut_usb_strerror(err)`. The original code was half-wrong too though :)

Reply:
You're right we shouldn't drop the `err` here, was a bit over-eager on my part, suggestion:
`upsdebugx(3, "USB: Send_USB_Packet: send_usb_packet, err = %d %s ", err, nut_usb_strerror(err));`
or...
`upsdebug_with_errno(3, "USB: Send_USB_Packet: send_usb_packet, err = %d %s ", err,  nut_usb_strerror(err));`
for also including the OS-based `errno` although probably kind of irrelevant here?

(applied to the other occurrences as well)

Reply:
I think the latter does make sense, to add context maybe (e.g. some generic error in lib and perms in OS)?..

Perhaps likewise in other USB code, if there still are cases...

Reply:
Good thought, will amend accordingly and look for similar occurrences elsewhere while at it.